.github/workflows/unit-test.yml

```diff
@@ -151,8 +151,8 @@ jobs:
         run: npm install --ignore-scripts
       - name: Bootstrap Dependencies
         run: npx lerna bootstrap --no-ci --hoist --nohoist='zone.js' --nohoist='mocha' --nohoist='ts-mocha'
-      - name: Unit tests
-        run: npm run test:ci:changed -- ${{ matrix.lerna-extra-args }}
+      - name: Unit Test
+        run: npm run test:ci -- --since ${{github.event.pull_request.base.sha || github.event.push.before }} ${{ matrix.lerna-extra-args }}
```

Comment:
>In this case, wouldn't running the tests with --since github.event.push.before will cause partial reports that only tracks the changed package in each merge to main, but will skip packages that have not had any change in the merge?

Yes, that also why I want to confirm this behavior with @pichlermarc 

Codecov also support Flag feature with CarryForward, you can see one of the report here:
https://app.codecov.io/gh/open-telemetry/opentelemetry-js-contrib/commit/7b14c92d6913d104401ec086034168dfc57aedfe
![image](https://github.com/open-telemetry/opentelemetry-js-contrib/assets/10696932/8ed0592b-bb58-438e-b949-9bb7d7012578)
In above commit, I configure to skip the node test but the codecov still capture the previous node's code coverage in single report.

But to fully support our codebase, each package must define it own flag/paths in codecov.yml which could become cumbersome.

>If I remember correctly, the"Unit test" CI step run time was few minutes, and not dramatic compared to overall workflow time. 

Yes, a full unit test is arround 5m, but it relative small compare to npm install + lerna bootstrap step which account for 30m of execution time.

Reply:
@blumamir I dig a little bit on to codecov's Components and Flags and think it could potential improve but would require more code change + some potential maintaince burden.  I can create a prototype PR for that.

So I think for now it better to just do a full-run on node@14 to collect coverage report. For Node@16 and node@18 we can run a delta-run to speed thing up a bit, what do you think?